src/calendarsearchmodel.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2023 Damien Caliste <dcaliste@free.fr>
+ *
+ * You may use this file under the terms of the BSD license as follows:
+ *
+ * "Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in
+ *     the documentation and/or other materials provided with the
+ *     distribution.
+ *   * Neither the name of Nemo Mobile nor the names of its contributors
+ *     may be used to endorse or promote products derived from this
+ *     software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE."
+ */
+
+#include "calendarsearchmodel.h"
+#include "calendarmanager.h"
+#include "calendareventoccurrence.h"
+#include "logging_p.h"
+
+#include <QDebug>
+
+CalendarSearchModel::CalendarSearchModel(QObject *parent)
+    : CalendarEventListModel(parent)
+{
+}
+
+CalendarSearchModel::~CalendarSearchModel()
+{
+    CalendarManager::instance()->cancelSearch(this);
+}
+
+QHash<int, QByteArray> CalendarSearchModel::roleNames() const
+{
+    QHash<int, QByteArray> roleNames = CalendarEventListModel::roleNames();
+    roleNames[SectionRole] = "section";
+    return roleNames;
+}
+
+QVariant CalendarSearchModel::data(const QModelIndex &index, int role) const
+{
+    if (role < SectionRole)
+        return CalendarEventListModel::data(index, role);
+
+    QVariant data = CalendarEventListModel::data(index, CalendarEventListModel::OccurrenceObjectRole);
+    if (!data.isValid())
+        return QVariant();
+
+    CalendarEventOccurrence *occurrence = qobject_cast<CalendarEventOccurrence*>(data.value<QObject*>());
+    if (!occurrence)
+        return QVariant();
+
+    switch (role) {
+    case SectionRole:
+        return QVariant(occurrence->startTime().date().year());
```

Comment:
Section is what this can get used for but the role could be more like "year".

Reply:
You're right. The role is now called `YearRole` with a string represention of `year`.